onnx/defs/reduction/defs.cc

```diff
@@ -59,19 +80,38 @@ False instead of True.)DOC";
         return;
       }
 
-      int64_t keep_dims = 1;
+      int64_t keep_dims = 1, noop_with_empty_axes = 0;
       auto attr_proto = ctx.getAttribute("keepdims");
       if (attr_proto) {
         keep_dims = attr_proto->i();
       }
+      auto noop_attr_proto = ctx.getAttribute("noop_with_empty_axes");
+      if (noop_attr_proto) {
+        noop_with_empty_axes = noop_attr_proto->i();
+      }
+      std::vector<int64_t> axes;
+      size_t num_inputs = ctx.getNumInputs();
+      if (num_inputs == 2) { // axes is input
+        const TensorProto* axesInitializer = ctx.getInputData(1);
```

Comment:
Check if the axes attribute is specified and produce an error-message in that case?